File: examples/python/guides/integrations/scraper_browserbase.py
# Third-party integration example - requires: pip install playwright browserbase
# See: /guides/web-scraping

import os

from browserbase import Browserbase
from playwright.async_api import async_playwright

bb = Browserbase(api_key=os.environ["BROWSERBASE_API_KEY"])


# > Browserbase usage
async def scrape_url(url: str) -> dict:
    session = bb.sessions.create(project_id=os.environ["BROWSERBASE_PROJECT_ID"])
    async with async_playwright() as pw:
        browser = await pw.chromium.connect_over_cdp(session.connect_url)
        page = browser.contexts[0].pages[0]
        await page.goto(url)
        content = await page.content()
        await browser.close()
    return {"url": url, "content": content}

File: examples/typescript/guides/integrations/scraper-browserbase.ts
// Third-party integration - requires: pnpm add @browserbasehq/sdk playwright
// See: /guides/web-scraping

import Browserbase from '@browserbasehq/sdk';
import { chromium } from 'playwright';

const bb = new Browserbase({ apiKey: process.env.BROWSERBASE_API_KEY! });

// > Browserbase usage
export async function scrapeUrl(url: string) {
  const session = await bb.sessions.create({
    projectId: process.env.BROWSERBASE_PROJECT_ID!,
  });
  const browser = await chromium.connectOverCDP(session.connectUrl);
  const page = browser.contexts()[0].pages()[0];
  await page.goto(url);
  const content = await page.content();
  await browser.close();
  return { url, content };
}
